[docs] Editing pass on clusters docs, removing legacy material and fixing style issues (#27816)
ray-project/ray: doc/source/cluster/kubernetes/user-guides/configuring-autoscaling.md

(kuberay-autoscaling)=

# KubeRay Autoscaling

This guide explains how to configure the Ray autoscaler on Kubernetes.
The Ray autoscaler is a Ray cluster process that automatically scales a cluster up and down based on resource demand.
The autoscaler does this by adjusting the number of nodes (Ray pods) in the cluster based on the resources required by tasks, actors or placement groups.

Note that the autoscaler only considers logical resource requests for scaling (i.e., those specified in ``@ray.remote`` and displayed in `ray status`), not physical machine utilization. If a user tries to launch an actor, task, or placement group but there are insufficient resources, the request will be queued. The autoscaler adds nodes to satisfy resource demands in this queue.
The autoscaler also removes nodes after they become idle for some time.
A node is considered idle if it has no active tasks, actors, or objects.

```{admonition} When to use Autoscaling?
Autoscaling can reduce workload costs, but adds node launch overheads and can be tricky to configure.

> We recommend starting with non-autoscaling clusters if you're new to Ray.

Reviewer: Unfortunately I think all of our quickstart examples and the cluster launcher's defaults do use autoscaling... maybe add a TODO to switch out the first examples with fixed-size configs.

### Upscaling and downscaling speed

If needed, you can also control the rate at which nodes should be added to or removed from the cluster. For applications with many short-lived tasks, you may wish to adjust the upscaling and downscaling speed to be more conservative.

Use the `RayCluster` CR's `autoscalerOptions` field to do so. The `autoscalerOptions` field
carries the following subfields:

**`upscalingMode`**: This controls the rate of Ray pod upscaling. The valid values are:
    - `Conservative`: Upscaling is rate-limited; the number of pending worker pods is at most the number
      of worker pods connected to the Ray cluster.
    - `Default`: Upscaling is not rate-limited.
    - `Aggressive`: An alias for Default; upscaling is not rate-limited.

**`idleTimeoutSeconds`** (default 60s): This is the number of seconds to wait before scaling down an idle worker pod. Worker nodes are considered idle when they hold no active tasks, actors, or referenced objects (either in-memory or spilled to disk).

### Configuring the autoscaler sidecar container

> The `autoscalerOptions` also provides some options for configuring the autoscaler itself. Usually, it should not be necessary to modify these options.

Reviewer 2: ```suggestion
The `autoscalerOptions` field also provides options for configuring the autoscaler container. Usually, it is not necessary to specify these options.
```

**`resources`**: The `resources` subfield of `autoscalerOptions` sets optional resource overrides
for the autoscaler sidecar container. These overrides
should be specified in the standard [container resource
spec format](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-v1/#resources).

> The default values are as indicated below:

Reviewer 2: picking at my own wording
```suggestion
The default values are indicated below:
```
